src/main/scala/rps/GameRepository.scala

```diff
@@ -3,21 +3,24 @@ package rps
 import scala.collection.concurrent.TrieMap
 
 import model._
+import java.util.concurrent.atomic.AtomicInteger
 
 trait GameRepository {
-  def save(play: Play): Unit
-  def read(): Option[Play]
+  def save(play: Play): Int
+  def read(userId: Int): Option[Play]
 }
 
 class InMemoryGameRepository extends GameRepository {
-  private val map = TrieMap.empty[String, Play]
+  private val map = TrieMap.empty[Int, Play]
```

Comment:
Offtopic (not realted to this PR): beware `TrieMap` `getOrElseUpdate` operation is not thread safe. In Banksealer we prefer `ConcurrentHashMap`. Ask Ema for details.

Reply:
As I understand this, `TrieMap.getOrElseUpdate` should be thread-safe now
(as of scala 2.11.6 [1]) /cc @Cardu

[1] https://github.com/scala/scala/pull/4319

Reply:
As far as I remember `TrieMap` doesn't have a `computeIfAbsent` equivalent. That makes the semantic a little odd. But leave it to @Cardu

Reply:
TrieMap is thread safe regarding it's own structure, the problem with this is that the expression that computes the value may be called multiple times if there are more than one actor that concurrently try to insert an element that was not present.

So you have to be really careful to not cause any side effect in the expression that you pass to this method, for example, we were creating a table on the database in the evaluation, and sometimes we had exception for `table already existing error`

This is also stated in a comment in scala source code: https://github.com/scala/scala/blob/7986814ead106fc7edf82eb792878ca6f056df44/src/library/scala/collection/concurrent/TrieMap.scala